### ISSUE TITLE
boostless branch doesn't compile on Ubuntu Maverick

### ISSUE THREAD

**Issue by wjwwood** [wjwwood/serial#13]
When I try to build serial (git:boostless:c3a8275) on Ubuntu Maverick, using the build with ROS option I get this:

```
∫ make
mkdir -p bin
cd build && cmake -Wdev -DCMAKE_TOOLCHAIN_FILE=`rospack find rosbuild`/rostoolchain.cmake  ..
Building with ROS
[rosbuild] Building package serial
[rosbuild] Including /opt/ros/electric/stacks/ros_comm/clients/rospy/cmake/rospy.cmake
[rosbuild] Including /opt/ros/electric/stacks/ros_comm/clients/roslisp/cmake/roslisp.cmake
[rosbuild] Including /opt/ros/electric/stacks/ros_comm/clients/cpp/roscpp/cmake/roscpp.cmake
-- Configuring done
-- Generating done
-- Build files have been written to: /home/william/atrv_ws/serial/build
cd build && make 
make[1]: Entering directory `/home/william/atrv_ws/serial/build'
make[2]: Entering directory `/home/william/atrv_ws/serial/build'
make[3]: Entering directory `/home/william/atrv_ws/serial/build'
make[3]: Leaving directory `/home/william/atrv_ws/serial/build'
[  0%] Built target rospack_genmsg_libexe
make[3]: Entering directory `/home/william/atrv_ws/serial/build'
make[3]: Leaving directory `/home/william/atrv_ws/serial/build'
[  0%] Built target rosbuild_precompile
make[3]: Entering directory `/home/william/atrv_ws/serial/build'
make[3]: Leaving directory `/home/william/atrv_ws/serial/build'
make[3]: Entering directory `/home/william/atrv_ws/serial/build'
[ 20%] Building CXX object CMakeFiles/serial.dir/src/impl/unix.o
/home/william/atrv_ws/serial/src/impl/unix.cc: In member function ‘void serial::Serial::SerialImpl::reconfigurePort()’:
/home/william/atrv_ws/serial/src/impl/unix.cc:218: error: aggregate ‘serial_struct ser’ has incomplete type and cannot be defined
/home/william/atrv_ws/serial/src/impl/unix.cc:221: error: ‘baudrate’ was not declared in this scope
/home/william/atrv_ws/serial/src/impl/unix.cc:223: error: ‘ASYNC_SPD_MASK’ was not declared in this scope
/home/william/atrv_ws/serial/src/impl/unix.cc:224: error: ‘ASYNC_SPD_CUST’ was not declared in this scope
/home/william/atrv_ws/serial/src/impl/unix.cc:226: error: ‘buf’ was not declared in this scope
make[3]: *** [CMakeFiles/serial.dir/src/impl/unix.o] Error 1
make[3]: Leaving directory `/home/william/atrv_ws/serial/build'
make[2]: *** [CMakeFiles/serial.dir/all] Error 2
make[2]: Leaving directory `/home/william/atrv_ws/serial/build'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/home/william/atrv_ws/serial/build'
make: *** [all] Error 2
```

Not sure what is the deal here, a missing header?

**Comment by OmaManfred** [comment]
Hello wjwwood. I have exactly the same errors and viewed the fixes #13, added the lines that were different, (only the ifdef line), but no changes. I am trying to translate the xsens_driver for ROS from rospy to roscpp on linux. I didn't clone the full branch but got the serial.h / unix.h and the cc files..

**Comment by OmaManfred** [comment]
Oh, okay I got it. Since I didn't keep the folder structure I misinterpreted "# include <linux/serial.h>" as
serial/serial.h

Now it works.
